### PR TITLE
OpenHIM Platform performance testing

### PR DESCRIPTION
**Pull request by drizzentic**
## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive Grafana dashboard for monitoring performance metrics from k6 load testing.
	- Developed scripts for various performance testing scenarios: load, smoke, and volume testing for FHIR API endpoints.
	- Enhanced testing capabilities with a JSON bundle for simulating healthcare-related data.
	- Improved Prometheus and Grafana configurations for enhanced monitoring infrastructure.

- **Documentation**
	- Added a new section on "Performance Testing" in the documentation table of contents, linking to detailed guides.
	- Standardized formatting in the documentation for improved readability.

### CODE REVIEW

File: monitoring/grafana/dashboards/performance/k6.json [hunk 1]
```diff
@@ -0,0 +1,2041 @@
+{
```
[the remark this answers is not in the document]
Comment (drizzentic) [reply monitoring/grafana/dashboards/performance/k6.json:1]: This was left out intentionally. I will review the docs.

**Comment by drizzentic** [comment]
@rcrichton I have addressed the feedback and hooked up the result output to grafana. This is the sample output of the dashboard
image

File: test/performance/scripts/load.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+export const options = {
+  scenarios: {
+    load: {
+      executor: "constant-arrival-rate",
+      rate: 50,
+      timeUnit: "1s",
+      duration: "5m",
+      preAllocatedVUs: 50,
+      maxVUs: 200,
+    },
```
Comment (rcrichton) [inline test/performance/scripts/load.js:16]: I think we should adjust the test params to give us passing results with the platform in it's current state.

File: documentation/packages/README.md [hunk 3]
```diff
@@ -8,3 +8,25 @@ description: >-
 # 📦 Packages
 
 Package can be stood up individually using the `instant package init -n <package_name>` command, or they can be included in your own recipes. This can be accomplished by [creating a profile](https://app.gitbook.com/s/TwrbQZir3ZdvejunAFia/getting-started/config#launching-a-profile) that includes the necessary packages and any custom configuration packages.
+
+## Performance Testing
```
[the remark this answers is not in the document]
Comment (rcrichton) [reply documentation/packages/README.md:12]: In addition I'd like you to include the results of a run to give an indication of the current capabilities of platform. This should include the system tested on and the results achieved.

File: documentation/packages/README.md [hunk 4]
```diff
@@ -8,3 +8,25 @@ description: >-
...
+
+The performance scripts are located in the `test` folder. To run this script against a local or remote server, you can follow these steps:
```
Comment (rcrichton) [inline documentation/packages/README.md:14]: The test fodler from where? We need to specify the repository, user won't always have this cloned.

File: test/performance/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Building a k6 binary
+
+Example command to build a binary with the Faker extension:
+
+```
+docker run --rm -u "$(id -u):$(id -g)" -v "${PWD}:/xk6" grafana/xk6 build --with github.com/szkiba/xk6-faker --with github.com/grafana/xk6-output-prometheus-remote@latest
+```
+
+## Run a specific performance test
```
Comment (rcrichton) [inline test/performance/README.md:25]: I'm thinking we should also include a smoke test which runs just 1 transaction and ensures it succeeds. We should suggest in the docs that this is alway run first as some service need to be warmed up before they perform well.

**Review by rcrichton — CHANGES_REQUESTED** [review]
Thanks, some more comments.

File: documentation/packages/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (drizzentic) [reply documentation/packages/README.md:12]: Do we have a guides section? Also should we provide only sample results for a specific performance test?

[the remark this answers is not in the document]
Comment (rcrichton) [reply documentation/packages/README.md:12]: I think we need both. A guide on how to reproduce and the an example run using specific hardware to give people an idea of the current performance. So both.

The guides section is here: https://github.com/jembi/platform/tree/main/documentation/guides

Login to the gitbook app to get a better idea of how the docs are structured.

[the remark this answers is not in the document]
Comment (drizzentic) [reply documentation/packages/README.md]: I have made re-arrangements and added the test results in the guide. @rcrichton

**Review by rcrichton — APPROVED** [review]
LGTM, just one final comment. I don't see the content appearing in the gitbook content in the github check. I think you need to add the files path to `documentation/SUMMARY.md`

**Review by ItsMurumba — APPROVED** [review]
LGTM. Just need to resolve some conflicts `monitoring/docker-compose.ym`l

**Pull request merged**